src/test/java/ast/ASTNodesKtTest.kt

```diff
@@ -3,159 +3,95 @@ package ast
+        runAllTestInFolder(nodeTestCasesFolder)
     }
 
     @Test
-    fun `should have at least one sibling and then up`() {
-        // arrange
-        val inputAntlrString = """
-            A
-                B
-                C
-        """.trimIndent()
-        val tree = createParserTree(inputAntlrString)
-        val visitor = TreepatVisitorImplementation()
-        val root = visitor.visit(tree)
-        // act
-        countNodes(root)
-        // assert
-        assertEquals(countNode, 3)
-        // TODO - sibling should have only one node
-        assertTrue(countSibling >= 1)
+    fun `should run all child test cases`() {
+        val nodeTestCasesFolder = "./src/test/java/ast/child/"
+        runAllTestInFolder(nodeTestCasesFolder)
     }
-
     @Test
-    fun `should have many node, three children, two sibling and then up`() {
-        // arrange
-        val inputAntlrString = """
-            A
-                B
-                    C
-                D
-                E
-                    F
-                    G
-        """.trimIndent()
-        val tree = createParserTree(inputAntlrString)
-        val visitor = TreepatVisitorImplementation()
-        val root = visitor.visit(tree)
-        // act
-        countNodes(root)
-        // assert
-        assertEquals(countNode, 7)
-        assertEquals(countChild, 3)
-        // TODO - sibling should have only one node
-        assertTrue(countSibling >= 2)
+    fun `should run all sibling test cases`() {
+        val nodeTestCasesFolder = "./src/test/java/ast/sibling/"
+        runAllTestInFolder(nodeTestCasesFolder)
     }
-
     @Test
-    fun `should have one child not tab only one space`() {
-        // arrange
-        val inputAntlrString = """
-            A
-             B
-        """.trimIndent()
-        val tree = createParserTree(inputAntlrString)
-        val visitor = TreepatVisitorImplementation()
-        val root = visitor.visit(tree)
-        // act
-        countNodes(root)
-        // assert
-        assertEquals(countNode, 2)
-        assertEquals(countChild, 1)
+    fun `should run all mix test cases`() {
+        val nodeTestCasesFolder = "./src/test/java/ast/mix/"
+        runAllTestInFolder(nodeTestCasesFolder)
     }
 
     @Test
     fun `should have at least one sibling not end line`() {
+        // Old version
         // arrange
         val inputAntlrString = """
             A
                 B C
         """.trimIndent()
+        val output = """
+            A
+                B
+                C
+        """.trimIndent()
         val tree = createParserTree(inputAntlrString)
         val visitor = TreepatVisitorImplementation()
-        val root = visitor.visit(tree)
         // act
-        countNodes(root)
+        val root = visitor.visit(tree)
         // assert
-        assertEquals(countNode, 3)
+        assertEquals(root.toString(), output)
         // TODO - sibling should have only one node
```

Comment:
I think we then solved this `TODO`, should we delete it?

Reply:
No is solved yet.